src/com/facebook/buck/query/AbstractTargetLiteral.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2015-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.query;
+
+import com.facebook.buck.util.immutables.BuckStyleImmutable;
+
+import org.immutables.value.Value;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.concurrent.Executor;
+
+@BuckStyleImmutable
+@Value.Immutable
+abstract class AbstractTargetLiteral extends QueryExpression {
+
+  @Value.Parameter
+  abstract String getPattern();
+
+  @Override
+  public <T> Set<T> eval(QueryEnvironment<T> env, Executor executor)
+      throws QueryException, InterruptedException {
+    return env.getTargetsMatchingPattern(getPattern(), executor);
+  }
+
+  @Override
+  public void collectTargetPatterns(Collection<String> literals) {
+    literals.add(getPattern());
+  }
+
+  @Override
+  public String toString() {
+    // Keep predicate consistent with Lexer.scanWord!
```

Comment:
You're right, this piece is still copy+paste, and the Lexer was imported from Bazel with minor changes.
IIRC, unit tests are the only place TargetLiteral#toString() is used. I think this comment is a warning to keep this method consistent with the rules for quoting arguments, in order to compare the expected values properly. 
However, I now realize that the Lexer does not enforce all the rules mentioned in the comments (e.g. an unquoted word starting by '*' or '$' does not trigger any error).